Online-vs-Offline-Editors

```diff
@@ -0,0 +1,71 @@
+**Online Offline Editors**
+**There are several offline IDE that programmers use in order to stay more productive to learn and be more productive at work**
+
+1.Eclipse is more than a Java editor. Its biggest advantages are code completion
+by tabbing method which saves a lot of time writing documentations. 
+
+It has a build in syntax check to fix any mistype words while coding a project. All of
+features we've come to expect of an IDE are their code completion, templates, integration 
+with various SCMSs, and integration with build systems. 
+
+Its code formatting and cleanup tools are very powerful. I find its build system to work
+well and intuitively. I think these are the things upon which its reputation is really built. 
+
+Furthermore it has a refactoring that find and replace function, variables and classes of a specific work. 
+
+Eclipse offers more plugins, platforms, frameworks, and simpler to use compare 
+to other IDE such as IntelliJ, BlueJ and JEdit.
+
+2.NetBeans is a free intuitive editor that does it all compared from other IDE.
+It has simple Swing GUI design tool to build user interfaces by dragging and
+dropping components for instance buttons, textboxes. 
+
+The disadvantage it is that with all its features it is slow to load and uses 
+more memory than other IDE’s.
+
+3.IntelliJ IDEA it is the most efficient offline editor however it is not free.
+It is faster than most editors however it has numbers of problems. Some major
+advantages is using too much memory from system,
+
+4.BlueJ is a simple editor mostly used to teach java and object oriented
+programming. The main advantage of the editor over other ones is that it does
+not need a main method to run the program.
+
+**However there are several differences between online editors and the offline editors**
+
+1.CodeRun Studio is an online IDE that you build web application. It allows you to run and test code online. 
+
+The disadvantage is that there is a 14 days trial then it must be paid. It is difficult to save your project since
+you must pay in order to fully operate it.
+
+An offline IDE will be more beneficial to a programmer rather than an online
+IDE due to the fact that large projects will slow down and that a software
+team will need a reasonable office internet service. 
+
+If there is a major storm the connection will be affected causing many projects to be unfinished and
+incomplete. 
+
+The only advantages of online IDE is that projects can be access everywhere; there are no setup and configurations. 
+There is no hardware limitation since all information will be stores in cloud.
+
+Links:
+Eclipse - https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html
+Netbeans - https://netbeans.org/features/
+CodeRun - http://www.webappers.com/2009/12/04/coderun-studio-free-cross-platform-online-ide/
+IntelliJ Idea - https://www.jetbrains.com/idea/
```

Comment:
When you are using Markdown, use its features instead of writing MS Word style documents in plain text.